registry/storage/driver/azure/azure.go

```diff
@@ -483,6 +499,8 @@ func is404(err error) bool {
 	)
 }
 
+var _ storagedriver.FileWriter = &writer{}
```

Comment:
I am not sure we need this here as `newWriter` returns `storagedriver.FileWriter`. We do add this check in the `gcs` driver so we should maybe remove it from there.

Though we should probably prefer returning struct types and not interfaces 🤔

Reply:
Technically we don't need it, though it doesn't hurt to be explicit about intent.

> Though we should probably prefer returning struct types and not interfaces 🤔

In general, yes. In this instance, we have to return the interface-typed value for the driver to satisfy the `storagedriver.StorageDriver` interface.